### PR TITLE
Add web branch versioning

### PR DESCRIPTION
**Pull request by SYangster**
- change to deploy site from gh-pages branch
- allow switching of versions at `https://nvidia.github.io/NVFlare/version/{branch}` (keep main branch at `https://nvidia.github.io/NVFlare`, add version dropdown selector
- update github and internal links
- will cherry-pick to 2.5 after merged

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -31,28 +38,20 @@ jobs:
...
+      - name: Build project
         run: npm run build
+        env:
+          PUBLIC_GH_BRANCH: ${{ github.ref_name }}
+        working-directory: ${{ env.site_path }}
 
-      - name: Upload Pages Artifact
-        uses: actions/upload-pages-artifact@v3
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4.6.4
```
Comment (YuanTingHsieh) [inline .github/workflows/deploy.yml:52]: Is this better than the previous "actions/xxx"?

**Review by YuanTingHsieh — COMMENTED** [review]
One comment otherwise LGTM

File: web/src/components/tutorials.astro [hunk 2]
```diff
@@ -142,328 +144,328 @@ const tutorials = [
...
     description: "Step-by-step examples series with CIFAR-10 (image data) to showcase different NVIDIA FLARE features, workflows, and APIs.",
-    link: "https://github.com/NVIDIA/NVFlare/tree/main/examples/hello-world/step-by-step/cifar10"
+    link: `https://github.com/NVIDIA/NVFlare/tree/${gh_branch}/examples/hello-world/step-by-step/cifar10`
   },
   {
     title: "Step-by-Step HIGGS Examples",
     tags: ["beg.", "algorithm", "pandas", "sklearn", "xgboost"],
     description: "Step-by-step examples series with HIGGS (tabular data) to showcase different NVIDIA FLARE features, workflows, and APIs.",
-    link: "https://github.com/NVIDIA/NVFlare/tree/main/examples/hello-world/step-by-step/higgs"
+    link: `https://github.com/NVIDIA/NVFlare/tree/${gh_branch}/examples/hello-world/step-by-step/higgs`
```
Comment (YuanTingHsieh) [inline web/src/components/tutorials.astro:183]: Can this "example link" also be substituted with a variable? But we could do that in another PR if we want

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (SYangster) [reply .github/workflows/deploy.yml:52]: Yes this one is popular and useful for deploying from specific directories

File: web/src/components/tutorials.astro (hunk 2 above)
Comment (SYangster) [reply web/src/components/tutorials.astro:183]: which link are you referring to? These github links I've replaced the branch with a substitution
